### ISSUE TITLE
TypeErrors when creating collection (with and without slug variables)

### ISSUE THREAD

**Issue by HarleySalas** [payloadcms/payload-3.0-demo#172]
With any collection, when creating, or editing a document, I get one of two `Error: TypeError:` messages, depending on rather I have my `Collection` slug's value in a variable, or not; neither of which, I think should be happening.

with a variable for the `slug` field:
```
export const COLLECTION_SLUG_PAGE = 'page'

export const Page: CollectionConfig = {
  slug: COLLECTION_SLUG_PAGE,
  ```
I get the following error:
```
[11:56:29] ERROR: aborted
    err: {
      "type": "Error",
      "message": "aborted",
      "stack":
          Error: aborted
              at connResetException (node:internal/errors:787:14)
              at abortIncoming (node:_http_server:793:17)
              at socketOnClose (node:_http_server:787:3)
              at Socket.emit (node:events:530:35)
              at TCP.<anonymous> (node:net:337:12)
              at TCP.callbackTrampoline (node:internal/async_hooks:130:17)
      "code": "ECONNRESET"
    }
[11:56:29] ERROR: There was an error building form state
    err: {
      "type": "TypeError",
      "message": "Cannot read properties of undefined (reading 'split')",
      "stack":
          TypeError: Cannot read properties of undefined (reading 'split')
              at Object.buildFormState [as form-state] (webpack-internal:///(rsc)/./node_modules/.pnpm/@payloadcms+next@3.0.0-beta.21_@types+react@18.2.79_graphql@16.8.1_monaco-editor@0.48.0_next@_buu3htsurgtepppnlde5dnvvte/node_modules/@payloadcms/next/dist/routes/rest/buildFormState.js:76:47)
              at eval (webpack-internal:///(rsc)/./node_modules/.pnpm/@payloadcms+next@3.0.0-beta.21_@types+react@18.2.79_graphql@16.8.1_monaco-editor@0.48.0_next@_buu3htsurgtepppnlde5dnvvte/node_modules/@payloadcms/next/dist/routes/rest/index.js:537:55)
              at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
              at async C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\compiled\next-server\app-route.runtime.dev.js:6:53211
              at async e_.execute (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\compiled\next-server\app-route.runtime.dev.js:6:44492)
              at async e_.handle (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\compiled\next-server\app-route.runtime.dev.js:6:54465)
              at async doRender (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\base-server.js:1359:42)
              at async cacheEntry.responseCache.get.routeKind (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\base-server.js:1573:40)
              at async DevServer.renderToResponseWithComponentsImpl (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\base-server.js:1493:28)
              at async DevServer.renderPageComponent (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\base-server.js:1910:24)
              at async DevServer.renderToResponseImpl (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\base-server.js:1948:32)
              at async DevServer.pipeImpl (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\base-server.js:895:25)
              at async NextNodeServer.handleCatchallRenderRequest (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\next-server.js:272:17)
              at async DevServer.handleRequestImpl (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\base-server.js:791:17)
              at async C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\dev\next-dev-server.js:339:20 
              at async Span.traceAsyncFn (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\trace\trace.js:154:20)
              at async DevServer.handleRequest (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\dev\next-dev-server.js:336:24)
              at async invokeRender (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\lib\router-server.js:174:21)
              at async handleRequest (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\lib\router-server.js:353:24)
              at async requestHandlerImpl (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\lib\router-server.js:377:13)
              at async Server.requestListener (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\lib\start-server.js:141:13)
    }
```

without a variable for the `slug` field:
```
export const Page: CollectionConfig = {
  slug: 'page',
```
I get the following error.
```
[12:02:17] ERROR: There was an error building form state
    err: {
      "type": "TypeError",
      "message": "Cannot destructure property 'collectionSlug' of 'reqData' as it is undefined.",
      "stack":
          TypeError: Cannot destructure property 'collectionSlug' of 'reqData' as it is undefined.
              at Object.buildFormState [as form-state] (webpack-internal:///(rsc)/./node_modules/.pnpm/@payloadcms+next@3.0.0-beta.21_@types+react@18.2.79_graphql@16.8.1_monaco-editor@0.48.0_next@_buu3htsurgtepppnlde5dnvvte/node_modules/@payloadcms/next/dist/routes/rest/buildFormState.js:35:17)
              at eval (webpack-internal:///(rsc)/./node_modules/.pnpm/@payloadcms+next@3.0.0-beta.21_@types+react@18.2.79_graphql@16.8.1_monaco-editor@0.48.0_next@_buu3htsurgtepppnlde5dnvvte/node_modules/@payloadcms/next/dist/routes/rest/index.js:537:55)
              at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
              at async C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\compiled\next-server\app-route.runtime.dev.js:6:53211
              at async e_.execute (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\compiled\next-server\app-route.runtime.dev.js:6:44492)
              at async e_.handle (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\compiled\next-server\app-route.runtime.dev.js:6:54465)
              at async doRender (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\base-server.js:1359:42)
              at async cacheEntry.responseCache.get.routeKind (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\base-server.js:1573:40)
              at async DevServer.renderToResponseWithComponentsImpl (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\base-server.js:1493:28)
              at async DevServer.renderPageComponent (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\base-server.js:1910:24)
              at async DevServer.renderToResponseImpl (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\base-server.js:1948:32)
              at async DevServer.pipeImpl (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\base-server.js:895:25)
              at async NextNodeServer.handleCatchallRenderRequest (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\next-server.js:272:17)
              at async DevServer.handleRequestImpl (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\base-server.js:791:17)
              at async C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\dev\next-dev-server.js:339:20 
              at async Span.traceAsyncFn (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\trace\trace.js:154:20)
              at async DevServer.handleRequest (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\dev\next-dev-server.js:336:24)
              at async invokeRender (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\lib\router-server.js:174:21)
              at async handleRequest (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\lib\router-server.js:353:24)
              at async requestHandlerImpl (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\lib\router-server.js:377:13)
              at async Server.requestListener (C:\dev\reproductions\payload-3\node_modules\.pnpm\next@14.3.0-canary.7_react-dom@18.3.0_react@18.3.0\node_modules\next\dist\server\lib\start-server.js:141:13)
    }
```

I don't *think* a reproduction is necessary. It should occur for every payload project in `beta.21`, as far as I can tell.

However, I have a resolution to never submit another issue without attached [reproduction](https://github.com/HarleySalas/payload-3-reproductions/tree/slug-type-error). 🤣

To ensure you get the error message, go to a document `create` or `edit` page and *refresh* the page.

**Comment by JessChowdhury** [comment]
@HarleySalas I can see this error appeared in `beta.20` - but only happens in this demo, the error doesn't occur when running any of test suites in Payload core.

**Comment by HarleySalas** [comment]
As far as I can tell, it's not *actually* breaking anything. All of my projects are started from the `payload-3.0-demo` repo and *all* of them experience this issue, so I guess the variable is in there somewhere.

**Comment by HarleySalas** [comment]
> @HarleySalas I can see this error appeared in `beta.20` - but only happens in this demo, the error doesn't occur when running any of test suites in Payload core. Still looking into this.

and I do apologize, I forgot to mention; this bug appeared for me, for the first time in `beta.21`

**Comment by JessChowdhury** [comment]
It seems this error started when we bumped the `next` canary version `14.3.0-canary.7 > 14.3.0-canary.28`

**Comment by JessChowdhury** [comment]
@HarleySalas okay this should be fixed by updating `next` to `14.3.0-canary.37` - I'll update our dependency on the demo repo.
